### PR TITLE
third home done

### PR DESCRIPTION
**Pull request by Oxunjon1029**

### CODE REVIEW

File: models/Subscription.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  lastDayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  dayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  Category: {
```
Comment (AudriusBaltusis) [inline models/Subscription.js]: `categoryId`

File: models/piggybankmodel.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  lastDayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  dayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  Category: {
```
Comment (AudriusBaltusis) [inline models/piggybankmodel.js]: `categoryId`

File: models/transaction.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  title: {
+    type: String,
+    required: true
+  },
+  description: String,
+  dateOfOperattion: {
+    type: Date,
+    default: () => Date.now()
+  },
+  Category: {},
```
Comment (AudriusBaltusis) [inline models/transaction.js]: Please remove empty property

File: models/piggybankmodel.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const mongoose = require("mongoose");
```
Comment (AudriusBaltusis) [inline models/piggybankmodel.js]: file name is not consistent

File: models/obligatoryPayment.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  lastDayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  dayOfPayment: {
+    type: Date,
+    default: () => Date.now()
+  },
+  Category: {
```
Comment (AudriusBaltusis) [inline models/obligatoryPayment.js]: `categoryId`

File: models/transaction.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+
+const Transaction = new mongoose.Schema({
+  Type: {
+
+  },
+  account_id: {
```
Comment (AudriusBaltusis) [inline models/transaction.js]: `accountId`

File: routes/account.js [hunk 7]
```diff
@@ -1,23 +1,43 @@
...
 
 
-router.put('/edit/:id', (req, res) => {
-
-  res.send("Updated succesfully");
+router.put(`/edit`, async (req, res) => {
+  try {
+    const account = await Account.findByIdAndUpdate(req.body._id, req.body, { new: true })
+    res.status(200).json(account);
+    // res.send("Updated succesfully");
```
Comment (AudriusBaltusis) [inline routes/account.js]: Please remove unused code

File: routes/subscription.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+const express = require("express");
+const router = express.Router();
+
+router.post('/add', (req, res) => {
+  console.log('req', req);
```
Comment (AudriusBaltusis) [inline routes/subscription.js]: Please remove `console.log`

File: server.js [hunk 9]
```diff
@@ -72,6 +77,27 @@ app.get('/posts', auth, (req, res) => {
   res.json(db.users.filter(user => user.role.toLowerCase() === "user"));
 })
 
+
+userCreateAdmin();
```
Comment (AudriusBaltusis) [inline server.js]: This functionality shouldn't be in `server.js`

File: models/piggybankmodel.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+const PiggyBannkSchema = new mongoose.Schema({
```
Comment (AudriusBaltusis) [inline models/piggybankmodel.js]: Typo in model name

File: models/Category.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+const mongoose = require("mongoose");
+
+const Category = new mongoose.Schema({
+  title:{
+    type:String,
+    required:true
+  },
+  Type:[
```
Comment (AudriusBaltusis) [inline models/Category.js]: Could you explain this structure?

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

**Comment by Oxunjon1029** [comment]
In models/Category.js
<https://github.com/Oxunjon1029/budget/pull/3#discussion_r830001628>:

> @@ -0,0 +1,20 @@
+const mongoose = require("mongoose");
+
+const Category = new mongoose.Schema({
+  title:{
+    type:String,
+    required:true
+  },
+  Type:[

Could you explain this structure?:

In this case, in the category schema , we have income and expense category.
So in order to get either income category or expense category, I created
Type property like that, but I don't know whether it's true or not . Could
you help with that case?

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]
Please add `coverage` to `.gitignore`, it shouldn't be included into PR

**Comment by Oxunjon1029** [comment]
Done.

**Comment by AudriusBaltusis** [comment]
Why this PR was closed?

**Comment by Oxunjon1029** [comment]
After adding coverage file , that happened , why is that ?

**Comment by Oxunjon1029** [comment]
Could you help me with that ?

**Comment by Oxunjon1029** [comment]
I can't touch the reopen button below . If you can , could you reopen the pull request . I don't know how it happened .

**Comment by Oxunjon1029** [comment]
I finally reopened the pull request . Could you review all the changes and
give me feedback?

**Pull request closed without merging**